### PR TITLE
ASPSH-1008: Initial POC

### PR DESCRIPTION
**Pull request by davidnquach**
Created a client for the oauth provider server. The client has two parts the
frontend and backend.

The backend uses the express framework to build an API that makes calls to the
oauth server to get access tokens.

The frontend is react and used to display the user details.

### CODE REVIEW

File: server/controllers/session.ctrl.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const axios = require('axios');
+
+const client = axios.create({
+  baseURL: 'http://localhost:8000/'
+});
+
+module.exports = {
+  createSession: (req, res, next) => {
+    console.log(req.body.code);
```
Comment (addriv) [inline server/controllers/session.ctrl.js:9]: You're keeping console.logs?, you should look into a logger like Bunyan. I have a rule set on eslint to not allow console outs. I usually use console for debugging.

Comment (davidnquach) [reply server/controllers/session.ctrl.js:9]: It was something I used for local development, it should be removed. I'm probably not going to log it to a file either.

Comment (addriv) [reply server/controllers/session.ctrl.js:9]: the logger doesn't have to log into a file. It can log to just stdout.

Comment (davidnquach) [reply server/controllers/session.ctrl.js:9]: Yeah it probably shouldn't do that either given the security issue of having an auth code displayed.

**Comment by davidnquach** [comment]
@addriv 

Removed the node modules in the server app

**Comment by davidnquach** [comment]
@addriv 
Interesting discussion about adding/removing node modules 

ref: https://stackoverflow.com/questions/18128863/should-node-modules-folder-be-included-in-the-git-repository

**Comment by addriv** [comment]
@davidnquach npm shrinkpack that they are talking about is like package-lock.json. The lock file didn’t used to come with npm. Yarn had this feature.

**Comment by davidnquach** [comment]
@addriv 

I was more interested in the pros and cons of adding your node modules to your project, there seems to be arguments for both. So it depends on the type of project it is.

**Pull request closed without merging**